### PR TITLE
Add default values

### PR DESCRIPTION
**Pull request by gpotter2**
see https://github.com/daattali/beautiful-jekyll/pull/590

### CODE REVIEW

**Comment by daattali** [comment]
Thanks! There are still instances of config variables unaccounted for in the code, for example I know for certain that date_format is also in `_includes/header.html`. The codebase needs to be fully scanned for all config variables

**Comment by gpotter2** [comment]
You're right. I'll grep to make sure i don't miss anything

**Comment by gpotter2** [comment]
I've grepped the project. This should be all of them.
The few i left do not need a default value

File: _includes/footer.html [hunk 1]
```diff
@@ -25,8 +25,10 @@
           {%- endfor -%}
         </ul>
       <p class="copyright text-muted">
+      {% if site.author %}
```
Comment (daattali) [inline _includes/footer.html:28]: This should be `if site.author.name`

It's possible to define `author` but not a subfield `name` , in which case a bullet point will show up

File: _includes/head.html [hunk 2]
```diff
@@ -3,15 +3,15 @@
...
+  <title>{% if page.use-site-title %}{{ site.title }} {{ site.title-separator | default: '-' }} {{ site.description }}{% elsif page.title %}{{ page.title }}{% else %}{{ site.title }}{% endif %}</title>
 
   <meta name="author" content="{{ site.author.name }}" />
 
   {% if page.subtitle %}
   <meta name="description" content="{{ page.subtitle }}">
   {% endif %}
 
-  <link rel="alternate" type="application/rss+xml" title="{{ site.title }} {{ site.title-separator }} {{ site.description }}" href="{{ '/feed.xml' | absolute_url }}" />
+  <link rel="alternate" type="application/rss+xml" title="{{ site.title | default: 'Untitled' }} {{ site.title-separator }} {{ site.description }}" href="{{ '/feed.xml' | absolute_url }}" />
```
Comment (daattali) [inline _includes/head.html:14]: `title-separator` needs a default here too

File: _includes/head.html [hunk 3]
```diff
@@ -3,15 +3,15 @@
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
   <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, viewport-fit=cover">
 
-  <title>{% if page.use-site-title %}{{ site.title }} {{ site.title-separator }} {{ site.description }}{% elsif page.title %}{{ page.title }}{% else %}{{ site.title }}{% endif %}</title>
+  <title>{% if page.use-site-title %}{{ site.title }} {{ site.title-separator | default: '-' }} {{ site.description }}{% elsif page.title %}{{ page.title }}{% else %}{{ site.title }}{% endif %}</title>
 
   <meta name="author" content="{{ site.author.name }}" />
```
Comment (daattali) [inline _includes/head.html:8]: this meta tag shouldn't be included if author name isn't available

File: _includes/header.html [hunk 4]
```diff
@@ -1,5 +1,7 @@
 <!-- TODO this file has become a mess, refactor it -->
 
+{% assign date_format = site.date_format | default: "%B %-d, %Y" %}
```
Comment (daattali) [inline _includes/header.html:3]: `tags.html` also has a date_format that needs a default

File: _includes/head.html (hunk 3 above)
Comment (gpotter2) [reply _includes/head.html:8]: I assumed if you use `use-site-title`, you had provided a title..

**Comment by daattali** [comment]
Fair

File: _includes/head.html (hunk 3 above)
Comment (gpotter2) [reply _includes/head.html:8]: I assumed using `use-site-title` meant there was title. Another check seems over the top

**Comment by gpotter2** [comment]
Whoops I didn't mean to delete my comment.
Will fix those in a minute

**Comment by daattali** [comment]
Message me when all done

File: _includes/footer.html (hunk 1 above)
Comment (gpotter2) [reply _includes/footer.html]: done

File: _includes/head.html (hunk 2 above)
Comment (gpotter2) [reply _includes/head.html]: done

File: _includes/head.html (hunk 3 above)
Comment (gpotter2) [reply _includes/head.html:8]: - if page-use-title is here it's enough
- done

File: _includes/header.html (hunk 4 above)
Comment (gpotter2) [reply _includes/header.html:3]: done

**Comment by gpotter2** [comment]
Should be done.

**Comment by daattali** [comment]
I added one small commit, but otherwise looks good!

**Pull request merged**